app/src/ui/create-branch/create-branch-dialog.tsx

```diff
@@ -150,6 +150,10 @@ export class CreateBranch extends React.Component<
         ),
       })
     }
+
+    if (nextProps.initialName.length > 0) {
+      this.checkBranchRules(nextProps.initialName)
+    }
```

Comment:
I found that if I use the "New Branch" button in the branch dropdown with an invalid name, the repo rules were not being run on that initial branch name and I could create the branch as long as I didn't modify the name again. :P (An unintentional bypass to repo rules)